src/DevTools/Extension/components/Shell/Shell.tsx

```diff
@@ -1,74 +1,120 @@
 import { TabKeys, shellStyleDefaults } from '../../../constants';
 import { useDevtoolsJotaiStoreOptions } from '../../../internal-jotai-store';
 import { useSelectedShellTab } from './atoms';
 import { AtomViewer } from './components/AtomViewer';
 import { ErrorBoundary } from './components/ErrorBoundary';
 import { Header } from './components/Header';
-import { ShellResizeBar } from './components/ShellResizeBar';
 import { TabsHeader } from './components/TabsHeader';
 import { TimeTravel } from './components/TimeTravel';
 import { shellStyles } from './styles';
 
+function areWeTestingWithJest() {
+  return process.env.JEST_WORKER_ID !== undefined;
+}
+
 export const Shell = () => {
   const [selectedShellTab, setSelectedShellTab] = useSelectedShellTab();
 
   const shellRef = useRef<HTMLDivElement>(null);
+  const moveableRef = useRef<Moveable>(null);
 
   // TODO move this to a custom hook
-  const { height } = useAtomValue(
+  const [shellStyle, setShellStyles] = useAtom(
     shellStylesAtom,
     useDevtoolsJotaiStoreOptions(),
   );
 
+  useEffect(() => {
+    moveableRef.current?.updateRect();
+  }, [shellStyle]);
+
   const handleOnTabChange = (value: TabKeys) => setSelectedShellTab(value);
   return (
-    <Tabs
-      keepMounted={false}
-      variant="default"
-      defaultValue={TabKeys.AtomViewer}
-      sx={shellStyles}
-      h={height}
-      mah={shellStyleDefaults.maxHeight}
-      ref={shellRef}
-      className="jotai-devtools-shell"
-      data-testid="jotai-devtools-shell"
-      id="jotai-devtools-shell"
-      value={selectedShellTab}
-      onTabChange={handleOnTabChange}
-    >
-      <ShellResizeBar shellRef={shellRef} />
-      <Header />
-      <ErrorBoundary>
-        <TabsHeader />
-        <Tabs.Panel
-          value={TabKeys.AtomViewer}
-          h="100%"
-          sx={{
-            overflow: 'hidden',
-            // Hide the overlap of this div's bg
-            borderBottomLeftRadius: '7px',
-            borderBottomRightRadius: '7px',
-          }}
-        >
-          <AtomViewer />
-        </Tabs.Panel>
-        <Tabs.Panel
-          value={TabKeys.TimeTravel}
-          h="100%"
-          sx={{
-            overflow: 'hidden',
-            // Hide the overlap of this div's bg
-            borderBottomLeftRadius: '7px',
-            borderBottomRightRadius: '7px',
+    <>
+      <Tabs
+        keepMounted={false}
+        variant="default"
+        defaultValue={TabKeys.AtomViewer}
+        sx={(theme) => {
+          return {
+            ...(typeof shellStyles === 'function'
+              ? shellStyles(theme)
+              : shellStyles),
+            transform: shellStyle.transform,
+          };
+        }}
+        mah={shellStyleDefaults.maxHeight}
+        maw={shellStyleDefaults.maxWidth}
+        mih={shellStyleDefaults.minHeight}
+        miw={shellStyleDefaults.minWidth}
+        h={shellStyle.height}
+        w={shellStyle.width}
```

Comment:
This feels a bit hacky, would it be a good idea to have two separate variables, one holds the shell styles, and the other holds the styles set by Movable. And we merge those two here 🤔

Reply:
Do you mean merging `shellStyleDefaults` and `shellStyle`?

Reply:
Yes! That and `shellStyles` being a function or object. Ideally we don't want the type to be polymorphic at runtime.

Perhaps something like this?

```ts
const mergedTabStyles = (theme) => {
  return {
    ...shellStyleDefaults,
    ...shellStyles(theme),
    ...movableStyles,
  };
};

// ... 
sx={mergedTabStyles}
// ... 
```